gdal/frmts/gtiff/geotiff.cpp

```diff
@@ -18244,6 +18244,12 @@ CPLErr GTiffDataset::SetMetadata( char ** papszMD, const char *pszDomain )
         }
     }
 
+    if( EQUAL(pszDomain, "xml:XMP") && papszMD && *papszMD )
+    {
+        int nTagSize = strlen(*papszMD) + 1;
```

Comment:
explicit static_cast<int> should be done.
Are you sure that the nul terminated byte must be written ? I didn't check the specification, but the autotest/gdrivers/data/byte_with_xmp.tif doesn't include a nul terminated byte for example

Reply:
About the test case, I don't see how can I test it.
The problem is that the metadata is stored in the GDALMetadata as well. So if I set the xml:XMP metadata, save the file, open it again (with GDAL) and read the metadata, it will be there anyway (even without my PR). The difference on this PR is that it is stored in the TIFFTAG_XMLPACKET tiff field.

I do not see how can I access directly to this field in python, not using "GetMetadata". Is there any option?